ADR/0033-enable-native-opentelemetry-tracing.md

```diff
@@ -0,0 +1,40 @@
+# 33. Enable Native OpenTelemetry Tracing
+
+Date started: 2024-02-27
+Date accepted: 2024-04-02
+
+## Status
+
+Accepted
+
+## Context
+
+Konflux is a tool under active development and, therefore, unforeseen issues may arise. A recent (at the time of writing this ADR) example is the [long running](https://github.com/redhat-appstudio/build-definitions/pull/856/checks?check_run_id=22307468968) [e2e-test](https://github.com/redhat-appstudio/build-definitions/blob/main/.tekton/tasks/e2e-test.yaml) in [Konflux’s build definitions](https://github.com/redhat-appstudio/build-definitions). Fixing and debugging such issues is not a trivial thing for Konflux’s developers. Additional data, metrics, telemetry and tracing are essential in enabling Konflux developers and SREs to come up with fixes.
+
+Tracing, in particular, enables a straightforward model for dealing with complex, distributed systems. It gives unique insight into a system’s execution, grouping functions together. These grouping functions can be critical for finding fields that correlate to some problem, and provide powerful insights to reduce the range of possible causes.
+
+Konflux users will also benefit from tracing, as they gain a more introspective view into their application’s build pipeline, enabling them to focus on effective and impactful corrective actions.
```

Comment:
Did you mean Konflux developers/SRE instead of users or are you anticipating that tracing data will be readily accessible to them? For reference, users generally don't have access to system logs or metrics and we don't encourage them to request access either. I think we should be cautious about granting access to traces by default and move that decision or suggestion out of this ADR.

Reply:
In that case I'll remove all "users" mentions from the ADR.

Reply:
Removed all mentions to "users".